### DISCUSSION TITLE
How do I get the author of a branch via Git command?

### DISCUSSION THREAD

**Discussion by JasperMCompetente** [community/community#36537] [New to GitHub]
How do I get the author of a branch via Git command?

**Comment by ibrahimmemonn** [comment]
A branch is nothing but a commit pointer. As such, it doesn't track metadata like "who created me." See for yourself. Try cat .git/refs/heads/<branch> in your repository.

That written, if you're really into tracking this information in your repository, check out [branch descriptions.](https://stackoverflow.com/a/11886179/877115) They allow you to attach arbitrary metadata to branches, locally at least.

Also [DarVar's answer below](https://stackoverflow.com/a/19135644/877115) is a very clever way to get at this information.

**Comment by DSDmark** [comment]
Hey @Realfer15,
I'm not sure, but it could be help you 

You need to call **APIs,** 
but first thing first, check out that:-

[**GitHub Docs**](https://docs.github.com/en/rest/branches/branches#get-a-branch)

 For example, how you get your branch author :-

```https://api.github.com/repos/{owner}/{repo}/branches/{branch}```

> https://api.github.com/repos/DSDmark/GoRun/branches/main

The above is my repo link, when you fetch data from that **URL** or may past that URL on browser, you can see the following information:-
![Screenshot from 2022-10-18 16-14-30](https://user-images.githubusercontent.com/85744816/196410378-72bf08d9-cbcf-4e23-a840-c06fd1340ee9.png)

And also I suggest you if you want to use author then use like this  ```commit.committer.author.name``` 
I think that is the right form.

**Comment by ossanna16** [comment]
Hey there @JasperMCompetente and thank you for asking a good question! I'm glad you found our community 🙂
